### PR TITLE
Software 0.31 - Task usage metrics documentation

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: software/task-usage-metrics.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+global:
+  taskUsageMetricsEnabled: true
+
+  populateHourlyTaskAuditMetrics:
+    # Populate the Software UI with the latest task usage metrics on the 57th minute of every hour https://crontab.guru/#57_*_*_*_*
+    schedule: "57 * * * *"
+
+houston:
+  cleanupTaskUsageData:
+	  olderThan: 90
```
Comment (jwitz) [inline software/task-usage-metrics.md:26]:
```suggestion
  cleanupTaskUsageData:
          olderThan: 90
```

File: software/task-usage-metrics.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+title: 'Set up task usage metrics on Astronomer Software'
+sidebar_label: 'Set up task usage metrics'
+id: task-usage-metrics
+description: Configure the Astronomer Software platform to record metrics about task usage.
+---
+
+Task usage metrics provide a high level view of your Airflow task runs and can help you quickly identify Deployments where more tasks a running or failing than expected.
```
Comment (jwitz) [inline software/task-usage-metrics.md:8]:
```suggestion
Task usage metrics provide a high level view of your Airflow task runs and can help you quickly identify Deployments where more tasks are running or failing than expected.
```

File: software/task-usage-metrics.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Enable task usage metrics on your Software installation
+
+To enable task usage metrics for your entire Astronomer installation, configure the following values in your `config,yaml` file:
+
+```yaml
+global:
+  taskUsageMetricsEnabled: true
+
+  populateHourlyTaskAuditMetrics:
```
Comment (jwitz) [inline software/task-usage-metrics.md:20]: Is there any reason for the user to use the `Daily` configuration if they already use the `Hourly` one?

File: software/task-usage-metrics.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    # Populate the Software UI with the latest task usage metrics on the 57th minute of every hour https://crontab.guru/#57_*_*_*_*
+    schedule: "57 * * * *"
+
+houston:
+  cleanupTaskUsageData:
+          olderThan: 90
```
Comment (jwitz) [inline software/task-usage-metrics.md:26]: Will all organizations want a different number here, or should we recommend 90 in most cases?

File: software/task-usage-metrics.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics, sending them to Astronomer Support, and displaying them in the Software UI.
+
+To manually send task usage metrics to Astronomer support instead, see [Export task usage metrics manually](#export-task-usage-metrics-manually).
```
Comment (kwcanuck) [inline software/task-usage-metrics.md:31]:
```suggestion
To manually send task usage metrics to Astronomer support, see [Export task usage metrics manually](#export-task-usage-metrics-manually).
```

File: software/task-usage-metrics.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+## Set up task usage metrics on a Deployment (_Required only for Astronomer Certified and Astro Runtime < 6.1_)
+
+Task usage metrics are generated by the Astronomer-maintained `airflow-analytics-plugin`. This plugin queries the Airflow metadata database and returns usage analytics at an HTTP endpoint. Deployments using Runtime 6.1 or later have this plugin installed by default.
```
Comment (kwcanuck) [inline software/task-usage-metrics.md:35]:
```suggestion
Task usage metrics are generated by the Astronomer-maintained `airflow-analytics-plugin`. This plug-in queries the Airflow metadata database and returns usage analytics at an HTTP endpoint. Deployments using Runtime 6.1 or later have this plug-in installed by default.
```

File: software/task-usage-metrics.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+---
+title: 'Set up task usage metrics on Astronomer Software'
+sidebar_label: 'Set up task usage metrics'
+id: task-usage-metrics
+description: Configure the Astronomer Software platform to record metrics about task usage.
+---
+
+Task usage metrics provide a high level view of your Airflow task runs and can help you quickly identify Deployments where more tasks a running or failing than expected.
+
+If your organization's usage of Astronomer is priced based on successful task runs, you must configure task usage metrics so that you can export figures about your task runs to Astronomer.
```
Comment (bote795) [inline software/task-usage-metrics.md:10]: @cmarteepants do we keep this paragraph?

File: software/task-usage-metrics.md (hunk 3 above)
Comment (bote795) [reply software/task-usage-metrics.md:20]: they dont need to set these up, only if they want to change at what minute it runs and the same thing for the daily

hourly collects hourly and then daily aggregates data from hourly

File: software/task-usage-metrics.md (hunk 4 above)
Comment (bote795) [reply software/task-usage-metrics.md:26]: 90 is the default i believe @rishkarajgi and we should stick with it unless they want to retain more info which means more information in the db

File: software/task-usage-metrics.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  populateHourlyTaskAuditMetrics:
+    # Populate the Software UI with the latest task usage metrics on the 57th minute of every hour https://crontab.guru/#57_*_*_*_*
+    schedule: "57 * * * *"
+
+houston:
+  cleanupTaskUsageData:
+          olderThan: 90
+```
+
+After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics, sending them to Astronomer Support, and displaying them in the Software UI.
```
[the remark this answers is not in the document]
Comment (bote795) [reply software/task-usage-metrics.md:29]: i think we can remove this, as this was removed cc @cmarteepants @rishkarajgi

File: software/task-usage-metrics.md (hunk 5 above)
Comment (bote795) [reply software/task-usage-metrics.md:31]: i think we can remove this, as this was removed cc @cmarteepants @rishkarajgi

File: software/task-usage-metrics.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+## Set up task usage metrics on a Deployment (_Required only for Astronomer Certified and Astro Runtime < 6.1_)
+
+Task usage metrics are generated by the Astronomer-maintained `airflow-analytics-plugin`. This plugin queries the Airflow metadata database and returns usage analytics at an HTTP endpoint. Deployments using Runtime 6.1 or later have this plugin installed by default.
+
+To install the plugin on a Deployment using Astronomer Certified or Astro Runtime earlier than 6.1:
+
+1. Add the following line to the `requirements.txt` file of your Astro project:
+
+    ```text
+    https://github.com/astronomer/airflow-analytics-plugin/releases/download/1.0.0/astronomer_analytics_plugin-1.0.0-py3-none-any.whl
```
Comment (bote795) [inline software/task-usage-metrics.md:42]: we will have to update the version number when the next release is done of it @rishkarajgi , what is the new number?

File: software/task-usage-metrics.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+## View task usage metrics
+
+Depending on your permissions, you can view task usage metrics for a given installation, Workspace, or Deployment from the Software UI.
+
+- View an installation's task usage: In the Software UI, go to **System Admin** > **Task Usage**. This page shows total task runs, as well as task runs per Workspace.
+- View a Workspace's task usage: In the Software UI, select a Workspace and go to **Task Usage**. This page shows the Workspace's total task runs, as well as task runs per Deployment.
+- View a Deployment's task usage: In the Software UI, select open a Workspace, select a Deployment in the Workspace, and go to **Task Usage**. This page shows the Deployment's total task runs, as well as task runs per day.
+
+## Export task usage metrics manually
```
[the remark this answers is not in the document]
Comment (bote795) [reply software/task-usage-metrics.md:56]: export is a future feature and has been removed.

File: software/task-usage-metrics.md (hunk 4 above)
Comment (jwitz) [inline software/task-usage-metrics.md:26]:
```suggestion
```

File: software/task-usage-metrics.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+1. In the Software UI, go to **System Admin** > **Task Usage**.
+2. Click **Export Table**.
```
Comment (jwitz) [inline software/task-usage-metrics.md:59]:
```suggestion
```

File: software/task-usage-metrics.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+  populateHourlyTaskAuditMetrics:
+    # Populate the Software UI with the latest task usage metrics on the 57th minute of every hour https://crontab.guru/#57_*_*_*_*
+    schedule: "57 * * * *"
+
+houston:
+  cleanupTaskUsageData:
+          olderThan: 90
+```
+
+After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics, sending them to Astronomer Support, and displaying them in the Software UI.
```
Comment (jwitz) [inline software/task-usage-metrics.md:29]:
```suggestion
After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics at the 57th minute of every hour, sending them to Astronomer support, and displaying them in the Software UI every. By default Astronomer installation retains this data for 90 days. 
```

File: software/task-usage-metrics.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  populateHourlyTaskAuditMetrics:
+    # Populate the Software UI with the latest task usage metrics on the 57th minute of every hour https://crontab.guru/#57_*_*_*_*
+    schedule: "57 * * * *"
+
+houston:
+  cleanupTaskUsageData:
+          olderThan: 90
+```
+
+After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics at the 57th minute of every hour, sending them to Astronomer support, and displaying them in the Software UI every. By default Astronomer installation retains this data for 90 days.
```
Comment (jwitz) [inline software/task-usage-metrics.md:29]:
```suggestion
After you set this configuration, Astronomer Software begins to run a cron job for collecting task usage metrics at the 57th minute of every hour, sending them to Astronomer support, and displaying them in the Software UI every. By default, your Astronomer installation retains this data for 90 days. 
```

**Comment by vishwas-astro** [comment]
We may not want to expose this doc just yet since the feature is not yet ready for primetime and is shipped inside of 0.31.0 in a disabled state only. cc: @cmarteepants

File: software/task-usage-metrics.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+## Enable task usage metrics on your Software installation
+
+To enable task usage metrics for your entire Astronomer installation, configure the following values in your `config,yaml` file:
+
+```yaml
+global:
+  taskUsageMetricsEnabled: true
+
+
```
Comment (jwitz) [inline software/task-usage-metrics.md:20]:
```suggestion
```

**Comment by jwitz** [comment]
@vishwas-astro If you anticipate that customers will still want to configure this in 0.31, we could release the documentation in an unindexed state. The doc would only be accessible from the URL, so the Astronomer team would have to share the doc with qualified/ ready customers in order for them to see it. What do you think?

File: software/task-usage-metrics.md (hunk 7 above)
Comment (cmarteepants) [reply software/task-usage-metrics.md:10]: No, let's remove the line 10. Line 8 is good though

**Comment by cmarteepants** [comment]
> We may not want to expose this doc just yet since the feature is not yet ready for primetime and is shipped inside of 0.31.0 in a disabled state only. cc: @cmarteepants @bote795 @rishkarajgi

@vishwas-astro Agreed that we should wait until we officially publish the fixed plugin and get a new set of runtime images published with the updated plugin. Hopefully that will happen shortly. 

@jedcunningham had agreed we could publish another round once the plugin was fixed. @bote795 or @rishkarajgi, please confirm when that happens so we can merge this into the docs (assuming ofc the docs are approved)

File: software/task-usage-metrics.md (hunk 9 above)
Comment (rishkarajgi) [reply software/task-usage-metrics.md:33]: 1.0.3 is the current version. 1.0.4 is scheduled next with some changes

**Comment by rishkarajgi** [comment]
@jwitz The updated analytics changes have been deployed as part of Runtime 7.2.0 [release](https://astronomer.slack.com/archives/C020PVBJPKK/p1674255408971059). This PR can now be merged
cc: @vishwas-astro

File: docusaurus.config.js [hunk 15]
```diff
@@ -98,10 +98,15 @@ module.exports = {
     softwareNav: {
       items: [
         {
-          label: '0.30 (Latest)',
+          label: '0.31 (Latest)',
```
Comment (rishkarajgi) [inline docusaurus.config.js:101]: There was a bug fix pushed as part of 0.31.2 release. We might want to update the latest release to 0.31.2

**Review by rishkarajgi — CHANGES_REQUESTED** [review]
Need to address some comments on the updated versions

**Comment by jwitz** [comment]
@rishkarajgi There's so many merge conflicts here that I'm going to just close this PR and reopen a PR where we can focus on the specific doc. I will make the remaining changes needed between PRs!

**Pull request closed without merging**